File: 7. Manejo de errores y depuración/JavaScript/errores.js
// 1. Crea un script en JavaScript con un error de tipo y uno de referencia, y muestra cómo corregirlos.
// 2. Escribe un código con un error lógico (por ejemplo, un bucle infinito) y explica cómo se puede depurar.

// Error de tipo
let x = 10;
let y = "10";
console.log(x + y); // 1010

// Error de referencia
let z;
console.log(z); // undefined

// Error lógico
let i = 0;
while (i < 10) {
  console.log(i);
  // i++;
}

// Para depurar el error lógico, se puede utilizar un debugger o un console.log para ver el valor de la variable i en cada iteración del bucle. También se puede agregar una condición de salida para evitar el bucle infinito. Por ejemplo:
let j = 0;
while (j < 10) {
  console.log(j);
  j++;
}

File: 7. Manejo de errores y depuración/Python/errores.py
# 1. Escribe un programa en Python que genere un error de sintaxis y uno de tiempo de ejecución, y luego explica cómo podrían ser solucionados.
# 2. Crea un programa que contenga un error lógico y explica cómo identificarlo durante la ejecución.

# Error de sintaxis
# print("Hola Mundo") # Error de sintaxis, falta cerrar comillas
# Solución: Agregar una comilla al final de la cadena de texto
print("Hola Mundo")

# Error de tiempo de ejecución
# a = 10
# b = 0
# print(a/b) # Error de tiempo de ejecución, división por cero
# Solución: Cambiar el valor de b a un número distinto de cero
a = 10
b = 2
print(a / b)

# Error lógico
# a = 10
# b = 20
# c = a + b # Error lógico, debería ser a - b

# Identificación del error lógico
# Se puede identificar el error lógico al observar que el resultado de la operación no es el esperado
a = 10
b = 20
c = a - b  # Corrección del error lógico
print(c)
